### PR TITLE
Sleep sort in python

### LINKED ISSUE #2018: Sleep sort in Java
## 🚀 Feature

[DWOC Contribution]
Sleep sort in Java

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes

## Motivation
Sleep sort works by starting a separate task for each item to be sorted, where each task sleeps for an interval corresponding to the item's sort key, then emits the item. Items are then collected sequentially in time.

## Pitch
(Please explain why this feature should be implemented and how it would be used.)

### PR DESCRIPTION
**Pull request by sakshikhachane**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)? 

Yes

### Description

Added Sleep sort in python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes: #2018

### CODE REVIEW

**Comment by sakshikhachane** [comment]
@atarax665 Please review the PR.

File: Python/sort/Sleep_sort.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Time Complexity -
+    1. Best case - O(NlogN + max(input))
+    2. Average case - O(NlogN + max(input))
+    3. Worst case - O(NlogN + max(input))
+    where N = number of elements in the input array, and array = input array elements
+    
+Space Complexity - O(N)
+
+'''
```
Comment (atarax665) [inline Python/sort/Sleep_sort.py]: Add a blank line at the EOF.

File: Python/sort/Sleep_sort.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+
+#    Sleep sort is a sorting algorithm used to sort non-negative integers. Sleep sort works by creating a separate thread for each item in the input which is to be sorted and then each thread sleeps for an amount of time which is proportional to the value of corresponding array element. Therefore, the thread which is having the least amount of sleep time wakes up first and the element gets printed and then the second least element and so on. The largest element wakes up after a long time and then the element gets printed at the last and thus the output obtained is a sorted array.
```
Comment (atarax665) [inline Python/sort/Sleep_sort.py]: Use doc string here.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/sort/Sleep_sort.py (hunk 1 above)
Comment (sakshikhachane) [reply Python/sort/Sleep_sort.py]: Done

File: Python/sort/Sleep_sort.py (hunk 2 above)
Comment (sakshikhachane) [reply Python/sort/Sleep_sort.py]: Done

**Comment by sakshikhachane** [comment]
@atarax665 I have done all the changes that were requested.

File: Python/sort/Sleep_sort.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Start all Threads
+    for i in thread_array:
+        i.start()
+
+    # Wait for all threads to terminate
+    for i in thread_array:
+        i.join()
+
+
```
Comment (atarax665) [inline Python/sort/Sleep_sort.py:44]: Put the driver code inside ```if __name__ == "__main__"``` block.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/sort/Sleep_sort.py (hunk 3 above)
Comment (sakshikhachane) [reply Python/sort/Sleep_sort.py:44]: @atarax665 Done. Please review the code now.

**Review by atarax665 — APPROVED** [review]

**Comment by sakshikhachane** [comment]
@atarax665 Are there any changes to be done?

**Comment by sakshikhachane** [comment]
@atarax665 @HarshCasper Is there any change that is to be done? If not please merge this PR.

**Review by atarax665 — APPROVED** [review]

**Pull request merged**